### PR TITLE
Implement secure key release

### LINKED ISSUE #14892: Support key export from Key Vault
For Key Vault 7.2-preview, we need to add support to export keys. This would be a method like `KeyClient.ExportKey` that takes the key name, key version (both required), and a body with the `KeyExportParameters` model found in the link below. This only has a single required property named `environment` (model property: `env`), so we should take this as a parameter. This returns a `KeyVaultKey` (model `KeyBundle`).

This requires a number of other changes:

- [x] Add optional `Exportable` boolean property to `KeyProperties`.
- [x] Add `ReleasePolicy` property to `KeyProperties` of type `KeyReleasePolicy`.
- [x] Add `PublicExponent` integer property to `CreateKeyOptions`.
- [x] Add "export" to `KeyOperation` enum. This will correctly impact the `CreateKeyOptions` class and the `KeyClient.UpdateKey` method..
- [x] Add `ReleasePolicy` property to `CreateKeyOptions`.
- [x] Add `ReleasePolicy` property to `ImportKeyOptions`.

See the [service version 7.1 -> 7.2-preview diff](https://portal.azure-devex-tools.com/tools/diff/eyJzb3VyY2VVcmwiOiJodHRwczovL2dpdGh1Yi5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvYmxvYi9tYXN0ZXIvc3BlY2lmaWNhdGlvbi9rZXl2YXVsdC9kYXRhLXBsYW5lL3JlYWRtZS5tZCIsInRhcmdldFVybCI6Imh0dHBzOi8vZ2l0aHViLmNvbS9BenVyZS9henVyZS1yZXN0LWFwaS1zcGVjcy9ibG9iL21hc3Rlci9zcGVjaWZpY2F0aW9uL2tleXZhdWx0L2RhdGEtcGxhbmUvcmVhZG1lLm1kIiwic291cmNlVGFnIjoicGFja2FnZS03LjEiLCJ0YXJnZXRUYWciOiJwYWNrYWdlLTcuMi1wcmV2aWV3In0=) 🔒 for details.

### PR DESCRIPTION
**Pull request by heaths**
Resolves #14892 sans tests; see #16789 and #16792

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs [hunk 1]
```diff
@@ -1106,5 +1111,110 @@ public virtual async Task<Response<KeyVaultKey>> ImportKeyAsync(ImportKeyOptions
...
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+            Argument.AssertNotNullOrEmpty(environment, nameof(environment));
+
+            using DiagnosticScope scope = _pipeline.CreateScope($"{nameof(KeyClient)}.{nameof(ExportKey)}");
+            scope.AddAttribute("key", name);
+            scope.Start();
+
+            try
+            {
+                return _pipeline.SendRequest(RequestMethod.Post, new KeyExportParameters(environment), () => new KeyVaultKey(name), cancellationToken, KeysPath, name, "/", version, "/export");
```
Comment (christothes) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs:1176]: is it worth extracting `"/"` and `"/export"` into consts for both methods?

**Review by christothes — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs:1176]: Not really. We hadn't elsewhere, and consts will just be "stamped" in place by the compiler anyway. The compiled result is no different. I think this is more readable, and it's what generated code does as well (generated code uses very few constants).

**Comment by heaths** [comment]
FWIW, I did write some tests that failed because the service hasn't implemented it as spec'd in the swagger, but I did examine the test recordings and request/response models were correct as spec'd. @herveyw-msft do you want us to go ahead and check this in, then? /cc @vcolin7

**Comment by heaths** [comment]
I spoke with Vick offline and, yes, we will merge this. It may not work for a while, but should eventually.

**Comment by heaths** [comment]
@vcolin7 could you at least review https://github.com/Azure/azure-sdk-for-net/pull/16794/files#diff-a5a8d91de8e78a2e9da740239ab1cf5c214ad8e30567fe1820465113c9b425fa and make sure that looks like what you expected given our conversations?

File: sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs [hunk 2]
```diff
@@ -24,6 +26,7 @@ public partial class CreateRsaKeyOptions : Azure.Security.KeyVault.Keys.CreateKe
         public int? KeySize { get { throw null; } set { } }
         public Azure.Security.KeyVault.Keys.KeyType KeyType { get { throw null; } }
         public string Name { get { throw null; } }
+        public int? PublicExponent { get { throw null; } set { } }
```
Comment (vcolin7) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs:29]: Is public exponent only valid for RSA keys?

File: sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs [hunk 3]
```diff
@@ -13,8 +13,10 @@ public partial class CreateKeyOptions
         public CreateKeyOptions() { }
         public bool? Enabled { get { throw null; } set { } }
         public System.DateTimeOffset? ExpiresOn { get { throw null; } set { } }
+        public bool? Exportable { get { throw null; } set { } }
         public System.Collections.Generic.IList<Azure.Security.KeyVault.Keys.KeyOperation> KeyOperations { get { throw null; } }
         public System.DateTimeOffset? NotBefore { get { throw null; } set { } }
+        public Azure.Security.KeyVault.Keys.KeyReleasePolicy ReleasePolicy { get { throw null; } set { } }
```
Comment (vcolin7) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs:19]: Will other types of keys not use a `ReleasePolicy`? Also, I thought we were renaming `KeyVaultReleasePolicy` to `ReleasePolicy`, did you mean to rename the parameter but not the class itself?

File: sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs (hunk 2 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs:29]: Yes. The swagger models get reused a lot and shouldn't always, but it's understandably easier. If they ever add support for other algorithms that require a public exponent, we can add it to their specialized options classes.

File: sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs (hunk 3 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs:19]: Yes, just the property should be `ReleasePolicy`. The model/class is `KeyReleasePolicy`. That was in the issues I opened as well.

**Review by vcolin7 — APPROVED** [review]
LGTM!

**Pull request merged**